airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultAirbyteSource.java

```diff
@@ -150,4 +156,17 @@ public void cancel() throws Exception {
     }
   }
 
+  private void logInitialStateAsJSON(final WorkerSourceConfig sourceConfig) {
+    if (!logConnectorMessages) {
+      return;
+    }
+
+    try {
+      final String json = new ObjectMapper().writeValueAsString(sourceConfig.getState().getState());
```

Comment:
Could use the helper here as well.

Reply:
is there somewhere to confirm that we do this getState() twice before we pass it to the connector? This seems to lineup with what I see in the CDK where stream names are at the top level of the state JSON blob. But this is a pretty critical part of the platform <> connector interface